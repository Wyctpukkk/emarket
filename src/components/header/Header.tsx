import { Link } from 'react-router-dom';
import { useAppSelector } from '../../hooks/redux';
import { Product } from '../../models/IProduct';

const Header = () => {
  interface cartState {
    arrayOfItems: Product[];
    totalItems: number;
    totalPrice: number;
  }

  const state: cartState = useAppSelector((state) => state.cartReducer);
  const totalPrice = state.totalPrice;
  const totalItems = state.totalItems;

  return (
    <header className="header">
      <div className="wrapper">
        <article className="upheader">
          <ul className="upheader__contacts">
            <li className="upheader__place">
              <div className="upheader__place_img"></div>
              <span>г. Кокчетав, ул. Ж. Ташенова 129Б</span>
              <br />
              <span>(Рынок Восточный)</span>
            </li>
            <li className="upheader__email">
              <div className="upheader__email_img"></div>
              <span> opt.sultan@mail.ru</span>
              <br />
              <span>На связи в любое время</span>
            </li>
          </ul>
          <ul className="nav">
            <li className="nav__item">
              <a className="nav__link" href="#">
                О компании
              </a>
            </li>
            <li className="nav__item">
              <a className="nav__link" href="#">
                Доставка и оплата
              </a>
            </li>
            <li className="nav__item">
              <a className="nav__link" href="#">
                Возврат
              </a>
            </li>
            <li className="nav__item">
              <a className="nav__link" href="#">
                Контакты
              </a>
            </li>
          </ul>
        </article>
      </div>
      <span className="line"></span>
      <div className="wrapper">
        <article className="downheader">
          <Link to={'/admin'} className="downheader__logotype"></Link>
          <button className="downheader__catalog">
            Каталог
            <span className="downheader__catalog_icon"></span>
          </button>
          <div className="downheader__search">
            <input className="downheader__input" placeholder="Поиск..."></input>
            <button className="downheader__search-btn"></button>
            <div className="downheader__search-icon"></div>
          </div>
          <div className="downheader__support">
            <div className="downheader__support-text">
              <span>+7 (777) 490-00-91</span>
              <span>время работы: 9:00-20:00</span>
              <span>Заказать звонок</span>
            </div>
            <div className="downheader__support-img"></div>
          </div>
          <button className="downheader__price-list">
            Прайс-лист
            <span className="downheader__price-list_icon"></span>
          </button>
          <div className="downheader__cart">
            <Link to={'cart'} className="downheader__cart-link">
              <span className="downheader__cart-link_img"></span>
              <span className="downheader__cart-link_count">{totalItems}</span>
            </Link>
            <div className="downheader__cart-info">
              Корзина
              <br />
              <span>{totalPrice} ₸</span>
            </div>
          </div>
        </article>
      </div>
      <span className="line2"></span>
    </header>
  );
};

export { Header };
