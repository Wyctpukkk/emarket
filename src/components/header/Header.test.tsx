import { render, screen, fireEvent } from '@testing-library/react';
import * as reduxHooks from 'react-redux';
import { MemoryRouter } from 'react-router-dom';
import { Header } from './Header';

jest.mock('react-redux');

describe('Header', () => {
  test('Проверка соответствия количества товаров и цены в корзине + снэпшот компонента', () => {
    jest
      .spyOn(reduxHooks, 'useSelector')
      .mockReturnValue({ totalPrice: 500, totalItems: 2 });

    const view = render(
      <MemoryRouter>
        <Header />
      </MemoryRouter>
    );

    const totalItems = screen.getByTestId(/testAmountItems/i);
    const totalPrice = screen.getByTestId(/testAmountMoney/i);
    expect(totalItems.textContent).toEqual('2');
    expect(totalPrice.textContent).toEqual('500 ₸');
    expect(view).toMatchSnapshot();
  });

  test('Проверка роутинга в корзину и админ панель', () => {
    jest
      .spyOn(reduxHooks, 'useSelector')
      .mockReturnValue({ totalPrice: 500, totalItems: 2 });

    render(
      <MemoryRouter>
        <Header />
      </MemoryRouter>
    );

    const firstLink = screen.getByTestId(/testAdminLink/i);
    fireEvent.click(firstLink);
    expect(screen.getByTestId(/testAdminLink/i)).toBeInTheDocument();
    const secondLink = screen.getByTestId(/testCartLink/i);
    fireEvent.click(secondLink);
    expect(screen.getByTestId(/testCartLink/i)).toBeInTheDocument();
  });
});
