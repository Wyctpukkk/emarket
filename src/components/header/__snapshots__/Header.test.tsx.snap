// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header Проверка соответствия количества товаров и цены в корзине + снэпшот компонента 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <header
        class="header"
      >
        <div
          class="wrapper"
        >
          <article
            class="upheader"
          >
            <ul
              class="upheader__contacts"
            >
              <li
                class="upheader__place"
              >
                <div
                  class="upheader__place_img"
                />
                <span>
                  г. Кокчетав, ул. Ж. Ташенова 129Б
                </span>
                <br />
                <span>
                  (Рынок Восточный)
                </span>
              </li>
              <li
                class="upheader__email"
              >
                <div
                  class="upheader__email_img"
                />
                <span>
                   opt.sultan@mail.ru
                </span>
                <br />
                <span>
                  На связи в любое время
                </span>
              </li>
            </ul>
            <ul
              class="nav"
            >
              <li
                class="nav__item"
              >
                <a
                  class="nav__link"
                  href="#"
                >
                  О компании
                </a>
              </li>
              <li
                class="nav__item"
              >
                <a
                  class="nav__link"
                  href="#"
                >
                  Доставка и оплата
                </a>
              </li>
              <li
                class="nav__item"
              >
                <a
                  class="nav__link"
                  href="#"
                >
                  Возврат
                </a>
              </li>
              <li
                class="nav__item"
              >
                <a
                  class="nav__link"
                  href="#"
                >
                  Контакты
                </a>
              </li>
            </ul>
          </article>
        </div>
        <span
          class="line"
        />
        <div
          class="wrapper"
        >
          <article
            class="downheader"
          >
            <a
              class="downheader__logotype"
              data-testid="testAdminLink"
              href="/admin"
            />
            <button
              class="downheader__catalog"
            >
              Каталог
              <span
                class="downheader__catalog_icon"
              />
            </button>
            <div
              class="downheader__search"
            >
              <input
                class="downheader__input"
                placeholder="Поиск..."
              />
              <button
                class="downheader__search-btn"
              />
              <div
                class="downheader__search-icon"
              />
            </div>
            <div
              class="downheader__support"
            >
              <div
                class="downheader__support-text"
              >
                <span>
                  +7 (777) 490-00-91
                </span>
                <span>
                  время работы: 9:00-20:00
                </span>
                <span>
                  Заказать звонок
                </span>
              </div>
              <div
                class="downheader__support-img"
              />
            </div>
            <button
              class="downheader__price-list"
            >
              Прайс-лист
              <span
                class="downheader__price-list_icon"
              />
            </button>
            <div
              class="downheader__cart"
            >
              <a
                class="downheader__cart-link"
                data-testid="testCartLink"
                href="/cart"
              >
                <span
                  class="downheader__cart-link_img"
                />
                <span
                  class="downheader__cart-link_count"
                  data-testid="testAmountItems"
                >
                  2
                </span>
              </a>
              <div
                class="downheader__cart-info"
              >
                Корзина
                <br />
                <span
                  data-testid="testAmountMoney"
                >
                  500
                   ₸
                </span>
              </div>
            </div>
          </article>
        </div>
        <span
          class="line2"
        />
      </header>
    </div>
  </body>,
  "container": <div>
    <header
      class="header"
    >
      <div
        class="wrapper"
      >
        <article
          class="upheader"
        >
          <ul
            class="upheader__contacts"
          >
            <li
              class="upheader__place"
            >
              <div
                class="upheader__place_img"
              />
              <span>
                г. Кокчетав, ул. Ж. Ташенова 129Б
              </span>
              <br />
              <span>
                (Рынок Восточный)
              </span>
            </li>
            <li
              class="upheader__email"
            >
              <div
                class="upheader__email_img"
              />
              <span>
                 opt.sultan@mail.ru
              </span>
              <br />
              <span>
                На связи в любое время
              </span>
            </li>
          </ul>
          <ul
            class="nav"
          >
            <li
              class="nav__item"
            >
              <a
                class="nav__link"
                href="#"
              >
                О компании
              </a>
            </li>
            <li
              class="nav__item"
            >
              <a
                class="nav__link"
                href="#"
              >
                Доставка и оплата
              </a>
            </li>
            <li
              class="nav__item"
            >
              <a
                class="nav__link"
                href="#"
              >
                Возврат
              </a>
            </li>
            <li
              class="nav__item"
            >
              <a
                class="nav__link"
                href="#"
              >
                Контакты
              </a>
            </li>
          </ul>
        </article>
      </div>
      <span
        class="line"
      />
      <div
        class="wrapper"
      >
        <article
          class="downheader"
        >
          <a
            class="downheader__logotype"
            data-testid="testAdminLink"
            href="/admin"
          />
          <button
            class="downheader__catalog"
          >
            Каталог
            <span
              class="downheader__catalog_icon"
            />
          </button>
          <div
            class="downheader__search"
          >
            <input
              class="downheader__input"
              placeholder="Поиск..."
            />
            <button
              class="downheader__search-btn"
            />
            <div
              class="downheader__search-icon"
            />
          </div>
          <div
            class="downheader__support"
          >
            <div
              class="downheader__support-text"
            >
              <span>
                +7 (777) 490-00-91
              </span>
              <span>
                время работы: 9:00-20:00
              </span>
              <span>
                Заказать звонок
              </span>
            </div>
            <div
              class="downheader__support-img"
            />
          </div>
          <button
            class="downheader__price-list"
          >
            Прайс-лист
            <span
              class="downheader__price-list_icon"
            />
          </button>
          <div
            class="downheader__cart"
          >
            <a
              class="downheader__cart-link"
              data-testid="testCartLink"
              href="/cart"
            >
              <span
                class="downheader__cart-link_img"
              />
              <span
                class="downheader__cart-link_count"
                data-testid="testAmountItems"
              >
                2
              </span>
            </a>
            <div
              class="downheader__cart-info"
            >
              Корзина
              <br />
              <span
                data-testid="testAmountMoney"
              >
                500
                 ₸
              </span>
            </div>
          </div>
        </article>
      </div>
      <span
        class="line2"
      />
    </header>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
